src/syrupy/session.py

```diff
@@ -138,10 +140,11 @@ def remove_unused_snapshots(
         used_snapshot_files: "SnapshotFiles",
     ) -> None:
         for snapshot_file, unused_snapshots in unused_snapshot_files.items():
-            for snapshot_name in unused_snapshots:
-                self._serializers[snapshot_file].delete_snapshot_from_file(
-                    snapshot_file, snapshot_name
-                )
+            serializer = self._serializers.get(snapshot_file)
+            if serializer:
+                serializer.delete_snapshots_from_file(snapshot_file, unused_snapshots)
+            elif snapshot_file not in used_snapshot_files:
+                os.remove(snapshot_file)
```

Comment:
I thought this has been moved into the serializer?

Reply:
If the serializer has been removed, there needs to be a way to remove unused files. I forgot that before in your PR